### ISSUE TITLE
Is there a reason parallel_read_safe is set to False?

### ISSUE THREAD

**Issue by jdillard** [sphinx-contrib/matlabdomain#147]
Are there known reasons why [parallel_read_safe](https://github.com/sphinx-contrib/matlabdomain/blob/fe8bb907cbcf391693afca274d3aa590bac0bf70/sphinxcontrib/matlab.py#L837) is set to `False` as opposed to `True`, as done in https://github.com/sphinx-contrib/matlabdomain/pull/124? Or was it just the safest option to suppress the warning at the time? If possible, it would be nice to have this extension not block parallel building.

For reference, the PHP domain has their [parallel_read_safe](https://github.com/markstory/sphinxcontrib-phpdomain/blob/5e84452a416bd3221092254e24df7b08900ad34d/sphinxcontrib/phpdomain.py#L774) set to `True`.

**Comment by joeced** [comment]
No, there is no specific reason. I took the safe choice. We could try to see if it's a problem, I can always revert it, if it does not work :)

**Comment by jdillard** [comment]
Sounds good. I'll try to experiment with setting `parallel_read_safe` to `True` in a fork and report back (only anecdotal evidence, but a start I guess)

**Comment by zouhairm** [comment]
@jdillard - any updates on running with `parallel_read_safe=True` ?

**Comment by jdillard** [comment]
Well the testing didn't go well:

```
Exception occurred:
File "/path/to/python3.9/site-packages/sphinx/util/parallel.py", line 111, in terminate
  self._result_funcs.pop(tid)
  KeyError: 0
```

I haven't seen that happen before, and this is the closest related thing I could find based on the error message: https://github.com/sphinx-doc/sphinx/issues/9210

As mentioned before, this is just anecdotal so recommend other people try as well before jumping to any conclusions.
